### PR TITLE
Run cupy-benchmark on CI

### PR DESCRIPTION
**Pull request by emcastillo**
Adds a job that is triggered with `/test benchmark` and compares the current PR with the master branch.
Useful to detect performance regressions in doubtful PRs

### CODE REVIEW

**Comment by emcastillo** [comment]
https://ci.preferred.jp/cupy.linux.benchmark/97821/

**Comment by emcastillo** [comment]
PTAL

File: .pfnci/config.pbtxt [hunk 1]
```diff
@@ -433,6 +433,41 @@ configs {
...
+      gpu: 1
+    }
+    time_limit: {
+      seconds: 21600
+    }
+    environment_variables { key: "GPU" value: "1" }
+    command: ".pfnci/linux/main-flexci.sh benchmark"
+  }
+}
+
```
Comment (kmaehashi) [inline .pfnci/config.pbtxt:470]: This can be removed?

File: .pfnci/linux/run.sh [hunk 2]
```diff
@@ -38,6 +40,7 @@ set -eu
 
 main() {
   if (( $# < 2 )); then
+    echo $#
```
Comment (kmaehashi) [inline .pfnci/linux/run.sh:43]: Remove?

File: .pfnci/linux/main-flexci.sh [hunk 3]
```diff
@@ -37,8 +37,14 @@ gcloud auth configure-docker
 
 echo "Starting: "${TARGET}""
 echo "****************************************************************************************************"
-CACHE_DIR=/tmp/cupy_cache PULL_REQUEST="${pull_req}" "$(dirname ${0})/run.sh" "${TARGET}" cache_get build test 2>&1 | tee "${LOG_FILE}"
+
+STAGES="cache_get build test"
+if [[ "${TARGET}" == "benchmark" ]]; then
+    STAGES="cache_get build benchmark"
+fi
+BENCHMARK_DIR=/tmp/benchmark CACHE_DIR=/tmp/cupy_cache PULL_REQUEST="${pull_req}" "$(dirname ${0})/run.sh" "${TARGET}" "${STAGES}" 2>&1 | tee "${LOG_FILE}"
```
Comment (kmaehashi) [inline .pfnci/linux/main-flexci.sh:45]: Need to upload `BENCHMARK_DIR`?

**Review by kmaehashi — CHANGES_REQUESTED** [review]

File: .pfnci/linux/main-flexci.sh (hunk 3 above)
Comment (emcastillo) [reply .pfnci/linux/main-flexci.sh:45]: No, we won't be uploading but running both branches in the PR I am removing this references.

We may want to upload stuff in the future maybe

File: .pfnci/config.pbtxt (hunk 1 above)
Comment (emcastillo) [reply .pfnci/config.pbtxt:470]: I will be using this config for PRs, master vs released branch will use the other config and run daily

File: .pfnci/linux/main-flexci.sh (hunk 3 above)
Comment (kmaehashi) [reply .pfnci/linux/main-flexci.sh:45]: Ah understood, so we can check the result in the console output!

**Comment by kmaehashi** [comment]
LGTM! Let's merge once this test runs fine:
https://ci.preferred.jp/cupy.linux.benchmark.pr/97863/

**Comment by kmaehashi** [comment]
Ah the test already ran 😅 https://ci.preferred.jp/cupy.linux.benchmark.pr/97848/

**Pull request merged**